pages/docs/concepts/asyncapi-document/adding-messages.md

```diff
@@ -0,0 +1,142 @@
+
+Here is an example of a simple message:
+
+```yml
+channels:
+  user/signedup:
+    address: 'application/json'
+    messages:
+      userSignUp:
+        $ref: '#/components/messages/userSignUp'
+```
+
+This document defines a `user/signedup` channel where a `userSignUp` message can be made.
+
+## Using `oneOf` for multiple message types
+
+AsyncAPI supports the use of `oneOf` for messages under operations, allowing you to define multiple possible message types for a single operation.
+
+Here is a diagram showing the use of `oneOf`:
+
+```mermaid
+graph TD;
+    A[Operations] --> B[Messages]
+    B --> C[oneOf]
+    C --> D[Message Type 1]
+    C --> E[Message Type 2]
+
+    style C fill:#47BCEE,stroke:#47BCEE;
+```
+
+This diagram shows how AsyncAPI uses `oneOf` to define multiple message types for a single operation.
+
+Here is an example document of how `oneOf` permits the use of multiple message types for a single operation:
+
+```yml
+channel:
+$ref: '#/channels/userSignupReply'
+messages:
+  oneOf:
+      - $ref: '#/components/messages/userSignedUp'
+      - $ref: '#/components/messages/userSignedUpReply'
+```
+
+The above document shows a channel `userSignupReply` under which two messages can be sent or received: `userSignedUp` or `userSignedUpReply`.
+
+## Specifying `contentType` in messages
+
+The `contentType` field specifies the format of the payload. If it's not provided, the default payload format is `application/json`. However, it's recommended always to specify the contentType for clarity and to avoid potential misinterpretations.
+
+Here is a diagram showing how to specify `contentType` in messages:
+
+```mermaid
+graph TD;
+    A[AsyncAPI Message] --> B[Payload]
+    B -->|Default| C[application/json]
+    B -->|Specified contentType| D[Other Content Types]
+
+    style A fill:#47BCEE,stroke:#47BCEE;
+```
+
+The above diagram shows an AsyncAPI message consists of a payload, which can default to application/json but should ideally have a specified contentType for clarity and precision.
+
+The following code shows how `contentType` is added to the Message:
+
+```yml
+messageId: userSignup
+name: UserSignup
+title: User signup
+summary: Action to sign a user up.
+description: A longer description
+contentType: application/json
+```
+
+In this document, the `contentType` is specified as `application/json` for the `userSignup`.
+
+## Reusing components
+
+The components object in the AsyncAPI specification contains reusable objects, but they will only impact the API if they are specifically referred to outside the components object.
+
+Here is a diagram explaining how to reuse components:
+
+```mermaid
+flowchart TB
+
+subgraph Components Object
+  Component_1
+  Component_2
+  Component_3
+end
+
+subgraph API
+  API_implementation --> Component_1:Uses
+  API_implementation --> Component_2:Uses
+  API_implementation --> Component_3:Uses
+
+  style API_implementation fill:#47BCEE,stroke:#47BCEE;
+end
+```
```

Comment:
I'm confused by this diagram. What exactly did you try to convey here? 🤔

Reply:
I tried to explain the **reusing components** section here. Do you think I should change it? In what way it would be better in your opinion?